### ISSUE TITLE
RuntimeError: Trying to backward through the graph second time, but the buffers have already been freed. Please specify retain_variables=True when calling backward for the first time.

### ISSUE THREAD

**Issue by susht3** [pytorch/examples#151]
reinforcement_learning/reinforce.py

1. to store experience of policy gradient,  i will call finish_episode() more than once to random sample some experiences, which gets this error:
RuntimeError: Trying to backward through the graph second time, but the buffers have already been freed. Please specify retain_variables=True when calling backward for the first time.

2. autograd.backward(self.saved_actions, [None for _ in self.saved_actions])
   what does [None] means？

**Comment by jeasinema** [comment]
Yes, I'm partly agree with you, but with a small correction, the algorithm implemented should be A2C(Advantage Actor Critic).
